Foli/user session association

creppyfm opened the pull request:
Configured User objects to take providerId as userId. Created UserDTO class to return requested user information to client. Configured sessions to expire after 30 days. Migrated /user endpoint to UserController (endpoint is now /api/users/user).

Began documenting endpoints with OpenAPI Operation annotations. Created OpenAPI config file to configure Swagger documentation.

creppyfm committed:
Created OpenApiConfig to configure Swagger documentation. Began customizing endpoint documentation with annotations in controller classes.

creppyfm committed:
Configured User objects to take providerId as userId. Created UserDTO class to return requested user information to client. Configured sessions to expire after 30 days. Migrated /user endpoint to UserController (endpoint is now /api/users/user).

kensonjohnson commented on server/src/main/java/com/creppyfm/server/controller/UserController.java:
The namespace for "/api" is already taken by nginx to point to the backend server. This means that all requests to the server will already have /api in them. As this code stands, when the production server is running the route will become "/api/api/users/user". In your code you should treat "/api" as the root "/" of your server, as that is what it will be.

kensonjohnson commented on server/src/main/java/com/creppyfm/server/dto_model/UserDTO.java:
What is DTO? This file could be more declarative if you avoid abbreviations.

kensonjohnson commented on server/src/main/java/com/creppyfm/server/config/HttpSessionConfig.java:
Consider pulling 2592000 out to a constant set at the top of the file with a comment describing it's length and/or usage.

kensonjohnson reviewed (changes requested):
Great work. Some minor changes listed below. Feel free to ignore if needed.

creppyfm committed:
Renamed dto_model to data_transfer_object_model, along with it's relevant class. Added comment to describe literal passed to annotation in Http
SessionConfig (cannot pass constants to annotations.) Removed /api from all controller mappings.

kensonjohnson reviewed (approved):
Looks good! :shipit:
